src/hotspot/cpu/x86/c1_MacroAssembler_x86.cpp

```diff
@@ -134,9 +135,14 @@ void C1_MacroAssembler::unlock_object(Register hdr, Register obj, Register disp_
   verify_oop(obj);
 
   if (LockingMode == LM_LIGHTWEIGHT) {
-    movptr(disp_hdr, Address(obj, hdr_offset));
-    andptr(disp_hdr, ~(int32_t)markWord::lock_mask_in_place);
-    lightweight_unlock(obj, disp_hdr, hdr, slow_case);
+#ifdef _LP64
+    lightweight_unlock(obj, disp_hdr, r15_thread, hdr, slow_case);
+#else
+    // This relies on the implementation of lightweight_unlock knowing that it
+    // will clobber its thread when using EAX.
```

Comment:
This use of `EAX` is confusing when earlier in this function `rax` is used.

Reply:
Hm, I thought RAX was confusing when used in x86_32 code. But can try to clarify.

Reply:
Tried to reword and restructure the comments surrounding this a bit.